cmd/cluster-init/update_jobs.go

```diff
@@ -160,16 +164,69 @@ func generateSecretVolume(clusterName string) v1.Volume {
 	}
 }
 
-func generateContainer(image, clusterName string, extraArgs []string, extraVolumeMounts []v1.VolumeMount) v1.Container {
+func generateContainer(image, clusterName string, extraArgs []string, extraVolumeMounts []v1.VolumeMount, extraEnvVars []v1.EnvVar) v1.Container {
+	var env []v1.EnvVar
+	env = append(env, extraEnvVars...)
+	if clusterName == string(api.ClusterBuild01) || clusterName == string(api.ClusterBuild02) {
+		env = append(env, []v1.EnvVar{
+			{
+				Name: clusterName + "_id",
+				ValueFrom: &v1.EnvVarSource{
+					SecretKeyRef: &v1.SecretKeySelector{
+						Key: clusterName + "-id",
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: clusterName + "-dex-oidc",
+						},
+					},
+				},
+			},
+			{
+				Name: "github_client_id",
+				ValueFrom: &v1.EnvVarSource{
+					SecretKeyRef: &v1.SecretKeySelector{
+						Key: clusterName + "_github_client_id",
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "build-farm-credentials",
+						},
+					},
+				},
+			},
+			{
+				Name: "slack_api_url",
+				ValueFrom: &v1.EnvVarSource{
+					SecretKeyRef: &v1.SecretKeySelector{
+						Key: "url",
+						LocalObjectReference: v1.LocalObjectReference{
+							Name: "ci-slack-api-url",
+						},
+					},
+				},
+			},
+		}...)
+	}
+	if clusterName == string(api.ClusterVSphere) {
```

Comment:
or build01 or build02, don't dupe this block

Reply:
I thought the ordering in the list mattered.
Updated.